### PR TITLE
Add tilt button to run pytest, fix local dev config for pytest, minor improvements to til

### PR DESCRIPTION
**Pull request by iskhakov**
# What this PR does

Fixes for https://github.com/grafana/oncall-private/issues/2423

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)

### CODE REVIEW

File: engine/settings/helm.py
Comment (joeyorlando) [inline engine/settings/helm.py:1]: why these changes here?

File: engine/settings/prod_without_db.py [hunk 1]
```diff
@@ -47,6 +45,8 @@ def on_uwsgi_worker_exit():
 
 CELERY_TASK_ROUTES = celery_task_routes.CELERY_TASK_ROUTES
 
+CELERY_TASK_ROUTES = celery_task_routes.CELERY_TASK_ROUTES
```
[the remark this answers is not in the document]
Comment (iskhakov) [reply engine/settings/prod_without_db.py:48]: Thank you for checking, I've messed my previous PR https://github.com/grafana/oncall/pull/3926, and this typo happened during conflict resolution

File: engine/settings/helm.py
Comment (iskhakov) [reply engine/settings/helm.py:1]: Thank you for checking, I've messed my previous PR https://github.com/grafana/oncall/pull/3926, and this typo happened during conflict resolution

File: engine/apps/alerts/tests/test_alert_receiver_channel.py [hunk 2]
```diff
@@ -272,6 +275,14 @@ def test_create_missing_direct_paging_integrations(
 def test_create_duplicate_direct_paging_integrations(make_organization, make_team, make_alert_receive_channel):
     """Check that it's not possible to have more than one active direct paging integration per team."""
 
+    # MariaDB is not supported for this test ()
```
Comment (iskhakov) [inline engine/apps/alerts/tests/test_alert_receiver_channel.py:278]:
```suggestion
    # MariaDB is not supported for this test
```

File: Tiltfile [hunk 3]
```diff
@@ -61,7 +61,7 @@ local_resource(
 
 local_resource(
     "e2e-tests",
-    labels=["E2eTests"],
+    labels=["Tests"],
```
Comment (iskhakov) [inline Tiltfile:64]: Need to check how to pin this section on the top

File: Tiltfile [hunk 4]
```diff
@@ -106,6 +106,30 @@ cmd_button(
     icon_name="dangerous",
 )
 
+# Inspired by https://github.com/grafana/slo/blob/main/Tiltfile#L72
+pod_engine_pytest_script = '''
+set -eu
+# get engine k8s pod name from tilt resource name
+POD_NAME="$(tilt get kubernetesdiscovery "engine" -ojsonpath='{.status.pods[0].name}')"
+kubectl exec "$POD_NAME" -- pytest .
```
Comment (joeyorlando) [inline Tiltfile:114]:
```suggestion
kubectl exec "$POD_NAME" -- pytest . --ds=settings.ci-test
```

I'm wondering if there's a way we could also provide optional arguments to `pytest`? Similar to how we currently have [`make test-dev`](https://github.com/grafana/oncall/blob/dev/Makefile#L180-L182) and you can run something like this:
```bash
$ make test-dev ARGS="apps/api/tests/test_auth.py"
```

File: engine/settings/tilt-dev.py
Comment (joeyorlando) [inline engine/settings/tilt-dev.py:1]: why a separate settings file for `tilt`?

**Review by joeyorlando — APPROVED** [review]

**Comment by joeyorlando** [comment]
@iskhakov is this PR still relevant?

File: engine/settings/tilt-dev.py
Comment (iskhakov) [reply engine/settings/tilt-dev.py:1]: good point, decided to use settings/dev.py instead.

File: Tiltfile (hunk 3 above)
Comment (iskhakov) [reply Tiltfile]: changed to allTests

File: Tiltfile (hunk 4 above)
Comment (iskhakov) [reply Tiltfile]: Added, thank you for the idea
